ci: update pwsh to use custom shell that fails-fast

m3dwards opened the pull request:
Github by default sets [fail fast](https://docs.github.com/en/actions/writing-workflows/workflow-syntax-for-github-actions#exit-codes-and-error-action-preference) behaviour on pswh shell which means that if any powershell cmdlet fails the script will stop and exit. The problem is that this behaviour doesn't apply when calling native executables, it only applies to powershell cmdlets.

I think the safest thing is to whenever we use pwsh to enable `$PSNativeCommandUseErrorActionPreference = $true` which will also fail calling any exe that returns a non-zero exit code.

Technically the step `Adjust paths in test/config.ini` only uses cmdlets so this step will not benefit from this change but I feel like it's good practice to still enable this feature in case this script gets modified in the future to call an exe.

Here is a CI run that has a script that fails silently (look at Windows Native, VS 2022 -> Get tool information): https://github.com/m3dwards/bitcoin/actions/runs/15415032095/job/43375709475
And with this change applied, the script correctly fails: https://github.com/m3dwards/bitcoin/actions/runs/15416585565/job/43380685364

hebasto commented:
Concept ACK.

Defining a new custom `job.defaults.run.shell` for all Windows jobs allows us to revert f8619196ceb5c6a58125506d276d9515837f043a and use the same shell across all steps. This is especially convenient for reproducing failures locally.

hebasto reviewed (commented):
Left a couple of notes regarding the current implementation:

1. The output in the "Get tool information" step has changed:
```diff
--- old	2025-07-03 13:37:57.062194420 +0100
+++ new	2025-07-03 13:35:57.800242845 +0100
@@ -1,3 +1,15 @@
+
+Name                           Value
+----                           -----
+PSVersion                      7.4.10
+PSEdition                      Core
+GitCommitId                    7.4.10
+OS                             Microsoft Windows 10.0.20348
+Platform                       Win32NT
+PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0…}
+PSRemotingProtocolVersion      2.3
+SerializationVersion           1.1.0.1
+WSManStackVersion              3.0
 cmake version 3.31.6
 
 CMake suite maintained and supported by Kitware (kitware.com/cmake).
```

2. The exit code is not propagated to the action's status.

m3dwards commented:
@hebasto is there any reason we need Powershell at all? Can we just use bash (or another language) everywhere?

m3dwards committed:
ci: update pwsh to use custom shell that fails-fast

Github Actions does set ErrorActionPreference to Stop but this does not
apply when calling native executables. Use this custom shell that will
fail fast when any command fails.

maflcko commented:
I've implemented the concept of my idea in https://github.com/bitcoin/bitcoin/pull/34500

maflcko commented:
Actually, https://github.com/bitcoin/bitcoin/pull/34583 fully removes pwsh and also passes CI
